resources/lib/ipwww_radio.py

```diff
@@ -154,55 +157,25 @@ def GetEpisodes(url):
 
 def AddAvailableLiveStreamItem(name, channelname, iconimage):
     """Play a live stream based on settings for preferred live source and bitrate."""
-    
-    stream_bitrates = [128] #TODO add more bitrates
-    if int(ADDON.getSetting('radio_source')) == 1:
-        providers = [('ak', 'Akamai')]
-    elif int(ADDON.getSetting('radio_source')) == 2:
-        providers = [('llnw', 'Limelight')]
-    else:
-        providers = [('ak', 'Akamai'), ('llnw', 'Limelight')]
-    bitrate_selected = 0 #TODO int(ADDON.getSetting('live_bitrate'))
-    for provider_url, provider_name in providers:
-        # First we query the available streams from this website
-
-        url = 'http://a.files.bbci.co.uk/media/live/manifesto/audio/simulcast/hds/uk/high/%s/%s.f4m' % (provider_url, channelname)
-        html = OpenURL(url)
-        # Use regexp to get the different versions using various bitrates
-        match = re.compile('href="(.+?)".+?bitrate="(.+?)"').findall(html)
-        streams_available = []
-        for address, bitrate in match:
-            url = address.replace('f4m', 'm3u8')
-            streams_available.append((int(bitrate), url))
-        streams_available.sort(key=lambda x: x[0], reverse=True)
-
-        # Play the prefered option
-        if bitrate_selected > 0:
-            match = [x for x in streams_available if (x[0] == stream_bitrates[bitrate_selected])]
-            if len(match) == 0:
-                # Fallback: Use any lower bitrate from any source.
-                match = [x for x in streams_available if (x[0] in range(1, stream_bitrates[bitrate_selected - 1] + 1))]
-                match.sort(key=lambda x: x[0], reverse=True)
-
-            PlayStream(name, match[0][1], iconimage, '', '')
-        # Play the fastest available stream of the preferred provider
-        else:
-            PlayStream(name, streams_available[0][1], iconimage, '', '')
+    providers = [('ak', 'Akamai'), ('llnw', 'Limelight')]
+    location_qualities = {'uk' : ['sbr_vlow', 'sbr_low', 'sbr_med', 'sbr_high'], 'nonuk': ['sbr_vlow', 'sbr_low'] }
+    location_names = {'uk': 'UK', 'nonuk': 'International'}
+    location_settings = ['uk', 'nonuk']
+    quality_colours = {'sbr_vlow': 'red', 'sbr_low': 'orange', 'sbr_med': 'yellow', 'sbr_high': 'green'}
+    quality_bitrates = {'sbr_vlow': '48', 'sbr_low': '96', 'sbr_med': '128', 'sbr_high': '320'}
 
+    location = location_settings[int(ADDON.getSetting('radio_location'))]
 
-def AddAvailableLiveStreamItemHQ(name, channelname, iconimage):
-    """Play a live stream based on settings for preferred live source and bitrate."""
+    for provider_url, provider_name in providers:
+        qualities = location_qualities[location]
+        qualities.reverse()
+        for quality in qualities: #TODO add Setting
+            url = 'http://a.files.bbci.co.uk/media/live/manifesto/audio/simulcast/hls/%s/%s/%s/%s.m3u8' % (location, quality, provider_url, channelname)
```

Comment:
This URL looks particularly interesting. It might lead the way to correct HLS live steams for video.